tests/unit/submissionworker/test_worker.py

```diff
@@ -0,0 +1,200 @@
+
+        self.client.force_authenticate(user=self.user1)
+
+        self.input_file = SimpleUploadedFile(
+            "test_file.txt", b"file_content", content_type="text/plain")
+
+        self.submission = Submission.objects.create(
+            participant_team=self.participant_team,
+            challenge_phase=self.challenge_phase,
+            created_by=self.challenge_host_team.created_by,
+            status="submitted",
+            input_file=self.input_file,
+            method_name="Test Method",
+            method_description="Test Description",
+            project_url="http://testserver/",
+            publication_url="http://testserver/",
+            is_public=True,
+        )
+
+    def tearDown(self):
+        shutil.rmtree('/tmp/evalai')
+
+    def test_download_and_extract_file(self):
+        download_location = join(self.BASE_TEMP_DIR, "test_file.txt")
+        self.url = return_file_url_per_environment(self.submission.input_file.url)
+        download_and_extract_file(self.url, download_location)
+        self.assertTrue(os.path.isfile(download_location))
+        os.remove(download_location)
+
+    def test_create_dir(self):
+        directory = join(self.BASE_TEMP_DIR, "temp_dir")
+        create_dir(directory)
+        self.assertTrue(os.path.isdir(directory))
+        shutil.rmtree(directory)
+
+    def test_create_dir_as_python_package(self):
+        directory = join(self.BASE_TEMP_DIR, "temp_dir")
+        create_dir_as_python_package(directory)
+        self.assertTrue(os.path.isfile(join(directory, "__init__.py")))
+        shutil.rmtree(directory)
+
+    def test_return_file_url_per_environment(self):
+        self.url = "/test/url"
+        returned_url = return_file_url_per_environment(self.url)
+        self.assertEqual(returned_url, "http://testserver/test/url")
+
+    @mock_sqs()
+    def test_get_or_create_sqs_queue_for_existing_queue(self):
+        client = boto3.client(
+            "sqs",
+            endpoint_url=os.environ.get("AWS_SQS_ENDPOINT", "http://sqs:9324"),
+            region_name=os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+            aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"),
+            aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
+        )
+        client.create_queue(QueueName="test_queue")
+        get_or_create_sqs_queue("test_queue")
+        queue_url = client.get_queue_url(QueueName='test_queue')['QueueUrl']
+        self.assertTrue(queue_url)
+        client.delete_queue(QueueUrl=queue_url)
+
+    @mock_sqs()
+    def test_get_or_create_sqs_queue_for_non_existing_queue(self):
+        client = boto3.client(
+            "sqs",
+            endpoint_url=os.environ.get("AWS_SQS_ENDPOINT", "http://sqs:9324"),
+            region_name=os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+            aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"),
+            aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
```

Comment:
@KhalidRmb can we move the client object as a class variable ?

Reply:
Done!